documentation/manual/src/main/asciidoc/en-US/modules/redis.asciidoc

```diff
@@ -0,0 +1,1946 @@
+    // getters, setters ...
+}
+----
+
+Basket collection
+
+[source, JSON]
+----
+Key: Basket:davide_basket
+{
+  "owner" : "Davide",
+  "products" : [ "Beer", "Pretzel" ]
+}
+----
+
+Product collection
+[source, JSON]
+----
+Key: Product:Beer
+{
+  "name" : "Beer",
+  "description" : "Tactical nuclear penguin"
+}
+
+Key: Product:Pretzel
+{
+  "name" : "Pretzel",
+  "description" : "Glutino Pretzel Sticks"
+}
+----
+====
+
+.Unidirectional one-to-many using one collection per strategy with @OrderColumn
+====
+[source, JAVA]
+----
+@Entity
+public class Basket {
+
+    @Id
+    private String id;
+
+    private String owner;
+
+    @OneToMany
+    private List<Product> products = new ArrayList<Product>();
+
+    // getters, setters ...
+}
+
+@Entity
+public class Product {
+
+    @Id
+    private String name;
+
+    private String description;
+
+    // getters, setters ...
+}
+----
+
+Basket collection
+
+[source, JSON]
+----
+Key: Basket:davide_basket
+{
+  "owner" : "Davide"
+}
+----
+
+Product collection
+[source, JSON]
+----
+Key: Product:Pretzel
+{
+  "description" : "Glutino Pretzel Sticks"
+}
+Key: Product:Beer
+{
+  "description" : "Tactical nuclear penguin"
+}
+----
+
+Redis List Associations:Basket_Product:davide_basket
+[source, JSON]
+----
+Rows:
+[
+{
+  "products_name" : "Pretzel",
+  "products_ORDER" : 1
+},
+{
+  "products_name" : "Beer",
+  "products_ORDER" : 0
+}
+]
+----
+====
+
+A map can be used to represents an association,
+in this case Hibernate OGM will store the key of the map
+and the associated id.
+
+.Unidirectional one-to-many using maps with defaults
+====
+[source, JAVA]
+----
+@Entity
+public class User {
+
+    @Id
+    private String id;
+
+    @OneToMany
+    private Map<String, Address> addresses = new HashMap<String, Address>();
+
+    // getters, setters ...
+}
+
+@Entity
+public class Address {
+
+    @Id
+    private String id;
+    private String city;
+
+    // getters, setters ...
+}
+----
+
+[source, JSON]
+----
+Key: User:user_001
+{ 
+  "addresses" : [
+    { 
+      "addresses_KEY" : "work",
+      "addresses_id" : "address_001"
+    },
+    {
+      "addresses_KEY" : "home",
+      "addresses_id" : "address_002"
+    }
+  ]
+}
+----
+
+[source, JSON]
+----
+Key: Address:address_001
+{
+  "city" : "Rome"
+}
+----
+
+[source, JSON]
+----
+Key: Address:address_002
+{
+  "city" : "Paris"
+} 
+----
+====
+
+You can use @MapKeyColumn to rename the column containing the key of the map.
+
+.Unidirectional one-to-many using maps with @MapKeyColumn
+====
+[source, JAVA]
+----
+@Entity
+public class User {
+
+    @Id
+    private String id;
+
+    @OneToMany
+    @MapKeyColumn(name = "addressType")
+    private Map<String, Address> addresses = new HashMap<String, Address>();
+
+    // getters, setters ...
+}
+
+@Entity
+public class Address {
+
+    @Id
+    private String id;
+    private String city;
+
+    // getters, setters ...
+}
+----
+
+[source, JSON]
+----
+Key: User:user_001
+{ 
+  "addresses" : [
```

Comment:
I think a more natural mapping would be

    "addresses" : [
        { "work" : "address_001" },
        { "home" : "address_002" }
    ]

But the proposed mapping is the same we use for Mongo/Couch so it's alright. There is an OGM issue somewhere to alter the map mapping for document stores.